cmd/soroban-cli/src/commands/tx/sign.rs

```diff
@@ -0,0 +1,97 @@
+use std::io;
+
+// use crossterm::{
+//     event::{self, Event, KeyCode, KeyEvent, KeyModifiers},
+//     execute,
+//     terminal::{self, EnterAlternateScreen, LeaveAlternateScreen},
+// };
+use soroban_sdk::xdr::{self, Limits, TransactionEnvelope, WriteXdr};
+
+use crate::signer::{self, InMemory, Stellar};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    XdrArgs(#[from] super::xdr::Error),
+    #[error(transparent)]
+    Signer(#[from] signer::Error),
+    #[error(transparent)]
+    Config(#[from] super::super::config::Error),
+    #[error(transparent)]
+    StellarStrkey(#[from] stellar_strkey::DecodeError),
+    #[error(transparent)]
+    Xdr(#[from] xdr::Error),
+    #[error(transparent)]
+    Io(#[from] io::Error),
+    #[error("User cancelled signing, perhaps need to add -y")]
+    UserCancelledSigning,
+}
+
+#[derive(Debug, clap::Parser, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    /// Confirm that a signature can be signed by the given keypair automatically.
+    #[arg(long, short = 'y')]
+    yes: bool,
+    #[clap(flatten)]
```

Comment:
We should leave out the yes, and require the user to specifier the list of signers who should sign, using the same format as expressed here:
- https://github.com/stellar/soroban-cli/issues/1264

And then address interactivity at a later time holistically.

For starters in this PR we could allow just one signer to begin with, and defer to #1264 adding support for multiple.